### DISCUSSION TITLE
Documentation for more complicated psyclone scripts -> transformations

### DISCUSSION THREAD

**Discussion by LonelyCat124** [stfc/PSyclone#2714] [General]
I was asked today about how our documentation for developing new trasnformations is - I wasn't sure if we had any but I think this would also be valid for creating more complex psyclone scripts for specific tasks (things that may end up as transformations later).

What do we currently have in the development guide and should we add discussion there about things like symbol table lookups, how to perform transformation validation, etc. for NG-Arch users?

**Comment by arporter** [comment]
I don't think we have any. I think there's something on adding a new PSyIR node but I agree, it would make a lot of sense if we had some! Strangely, there's a lot on Transformations in the Developer Guide that looks like it should really be in the User Guide.
